### PR TITLE
Introduce additional product options

### LINKED ISSUE #377: Introduce "optional" product options
Right now, we only manage mandatory options, i.e which appear as **radio buttons**. 

However, some options may not be mandatory, for example "add a slice of bacon", "add cheese", "add sauce"... 
Such options should appear as **checkboxes**. 

To manage this, we need to modify the `ProductOption` class to know how the group should behave. 
A few possible implementations: 
- Add a boolean property `isMultiple` or `isOptional` (quite redundant)
- Add a range `minChoices` / `maxChoices`: if `minChoices = maxChoices = 1`, then this is a radio buttons
- Mimic how this is done in Symfony [`ChoiceType`](https://symfony.com/doc/3.4/reference/forms/types/choice.html#select-tag-checkboxes-or-radio-buttons)

⚠️ The "optional options" **should not belong to the variant**. 
See `createProductVariant` in [`RestaurantController`](https://github.com/coopcycle/coopcycle-web/blob/f5b02c8eb2382aeae6a371c0d21ab2b6ec66b012/src/AppBundle/Controller/RestaurantController.php#L89)

### PR DESCRIPTION
**Pull request by alexsegura**
Fixes #377 

capture d ecran 2018-06-15 a 15 22 27

### CODE REVIEW

File: src/AppBundle/Resources/translations/messages.fr.yml [hunk 1]
```diff
@@ -484,6 +484,8 @@ form.product_option.add_value.label: Ajouter une valeur
 form.product_option.values.title: Valeurs
 form.product_option_value.value.label: Valeur
 form.product_option_value.price.label: Prix
+form.product_option.additional.label: Cette option est un supplément
```
Comment (Atala) [inline src/AppBundle/Resources/translations/messages.fr.yml]: I don't think this text is clear enough.
"Cette option est facultative."

File: src/AppBundle/Resources/views/Restaurant/index.html.twig [hunk 2]
```diff
@@ -126,20 +126,34 @@
                   {{ option.name }}
                   {% if option.strategy == 'option' %} - {{ option.price|price_format }}€{% endif %}
                 </h4>
-                <div class="list-group" data-product-option>
+                <div class="list-group" {% if not option.additional %}data-product-option{% endif %}>
```
Comment (Atala) [inline src/AppBundle/Resources/views/Restaurant/index.html.twig]: I think you should add a text on the modal intended to the customer. Like "optional" next to the modifier's name.

File: src/AppBundle/Controller/RestaurantController.php [hunk 3]
```diff
@@ -344,8 +344,18 @@ public function addProductToCartAction($id, $code, Request $request)
 
             $optionValues = [];
             foreach ($options as $optionCode => $optionValueCode) {
-                $optionValue = $productOptionValueRepository->findOneByCode($optionValueCode);
-                $optionValues[] = $optionValue;
+
+                $optionValueCodes = [];
+                if (!is_array($optionValueCode)) {
```
Comment (GaspardPO) [inline src/AppBundle/Controller/RestaurantController.php]: Inverting the if / else will remove the "!" in the condition.

File: src/AppBundle/Resources/translations/messages.fr.yml (hunk 1 above)
Comment (alexsegura) [reply src/AppBundle/Resources/translations/messages.fr.yml]: The problem is it translates to "this option is optional" 😅

File: src/AppBundle/Resources/views/Restaurant/index.html.twig (hunk 2 above)
Comment (alexsegura) [reply src/AppBundle/Resources/views/Restaurant/index.html.twig]: I think the option name can be used in this case, for example "Would you like an extra?"

**Comment by alexsegura** [comment]
@Atala nice catch, the code was not working when there were _only_ additional options. This is fixed. 
Maybe some day we should wrap all of this in a React component, but I merge this for now.

**Pull request merged**